File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
#include <unordered_set>
#include <iostream>
class Solution {
public:
    int lengthOfLongestSubstring(std::string s) {
        std::unordered_set<char> charset;
        int max_value = 0;
        int L = 0;
        for(int R=0; R< s.size();++R){
            while( charset.count(s[R])){
                charset.erase(s[L]);
                ++L;
            }
            charset.insert(s[R]);
            max_value = std::max(max_value,R-L+1);
        }
        return max_value;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        dict = {}
        for i in nums:
            dict[i] = i+1
        max_count = 0 
        for x in dict:
            count = 0
            while x in dict:
                count+=1
                x+=1
            max_count = max(max_count, count)

        return max_count